R/summarize_weighted_production.R

```diff
@@ -145,7 +143,7 @@ summarize_unweighted_production <- function(data, ..., with_targets = FALSE) {
   } else {
     data %>%
       summarize(weighted_production = .data$production, .groups = "keep") %>%
-      ungroup(.data$technology, .data$tmsr, .data$smsp) %>%
+      ungroup(all_of(c("technology", "tmsr","smsp"))) %>%
```

Comment:
pretty sure you should be able to use literal strings here without the `all_of()` and `c()` (`ungroup()` takes `...`), which seems less complicated, e.g.
``` r
tibble(a = 1, b = 2, c = 3) %>% 
  group_by(a, b) %>% 
  ungroup("a","b")
#> # A tibble: 1 × 3
#>       a     b     c
#>   <dbl> <dbl> <dbl>
#> 1     1     2     3
```

Reply:
I agree that it's possible, I notice that you can actually do something similar with `select` (It doesn't complain if you pass it a normal vector anymore). 

In any case it's pretty unclear to me what the best practice approach is, so I'd rather just play it as safe as possible for now. 

(In this PR: https://github.com/2DegreesInvesting/r2dii.analysis/pull/1 , `ungroup()` calls `all_of` in some places)

Reply:
^ yes totally, I noticed that as well, that's why it was still a little unclear to me